plugin/pkg/upstream/upstream.go

```diff
@@ -25,8 +25,11 @@ func (u *Upstream) Lookup(ctx context.Context, state request.Request, name strin
 		return nil, fmt.Errorf("no full server is running")
 	}
 
+	size := state.Size()
+	do := state.Do()
 	req := new(dns.Msg)
 	req.SetQuestion(name, typ)
+	req.SetEdns0(uint16(size), do)
```

Comment:
Perhaps we should be retaining the entire OPT record here. IIUC, ideally upstream should act as if the client made the request itself, in which case the entire OPT record, if present,  would be in the request.

Reply:
this is a new request for CNAMEs usually (when coming out the file plugin). So we don't have an original to speak off

Reply:
It's not in `state.Req`?